### PR TITLE
FLYTE_INTERNAL_IMAGE should have higher precedence

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
NA

## Why are the changes needed?
When unionai is installed, it will ignore `FLYTE_INTERNAL_IMAGE`, and always use the default [flytekit image](https://github.com/flyteorg/flytekit/blob/739d379029a12428171bab601f7f7f61ef86ed43/flytekit/configuration/default_images.py#L21-L27).

That's because flytekit will call `get_plugin().get_default_image()`, and then `DefaultImages.find_image_for()`. Thus, it skip the FLYTE_INTERNAL_IMAGE.

## What changes were proposed in this pull request?
Move `os.getenv("FLYTE_INTERNAL_IMAGE")` to `find_image_for()`

## How was this patch tested?
unit test

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
NA

## Docs link
NA

### CODE REVIEW

**Comment by pingsutw** [comment]
cc @thomasjpfan

File: flytekit/configuration/default_images.py [hunk 1]
```diff
@@ -35,13 +35,16 @@ def default_image(cls) -> str:
...
 
-        default_image_str = os.environ.get("FLYTE_INTERNAL_IMAGE", cls.find_image_for())
-        return default_image_str
+        return cls.find_image_for()
 
     @classmethod
     def find_image_for(
         cls, python_version: typing.Optional[PythonVersion] = None, flytekit_version: typing.Optional[str] = None
     ) -> str:
+        default_image_str = os.getenv("FLYTE_INTERNAL_IMAGE")
```
Comment (thomasjpfan) [inline flytekit/configuration/default_images.py:44]: `FLYTE_INTERNAL_IMAGE` is not really documented. What is it used for?

Comment (pingsutw) [reply flytekit/configuration/default_images.py:44]: It's very useful when I need to override the default image for the task or the base image in the ImageSpec.

Comment (eapolinario) [reply flytekit/configuration/default_images.py:44]: can we move `FLYTE_INTERNAL_IMAGE` to a constant and leave this comment there?

File: flytekit/core/constants.py [hunk 2]
```diff
@@ -9,3 +9,6 @@
 
 START_NODE_ID = "start-node"
 END_NODE_ID = "end-node"
+
+# To override the default container image or the default base image in ImageSpec, set this environment variable.
```
Comment (eapolinario) [inline flytekit/core/constants.py:13]:
```suggestion
# If set this environment variable overrides the default container image and the default base image in ImageSpec.
```

File: flytekit/core/constants.py [hunk 3]
```diff
@@ -9,3 +9,6 @@
...
+FLYTE_INTERNAL_IMAGE = "FLYTE_INTERNAL_IMAGE"
```
Comment (thomasjpfan) [inline flytekit/core/constants.py:14]: The convention with the reset of the code base is `_ENV_VAR`: 

```suggestion
FLYTE_INTERNAL_IMAGE_ENV_VAR = "FLYTE_INTERNAL_IMAGE"
```

https://github.com/flyteorg/flytekit/blob/b98e9416aee298435fc01d8d78f962bea60fd062/flytekit/configuration/file.py#L17-L18

Comment (pingsutw) [reply flytekit/core/constants.py:14]: Good catch, thanks!

**Review by eapolinario — APPROVED** [review]

**Pull request merged**